src/main.js

```diff
@@ -129,13 +133,26 @@ $(document).ready(() => {
             if (isSidebarPinned()) toggleSidebar();
             else togglePin();
           } else if (isSidebarVisible()) {
+            // @TODO(add current working branch to local store for a certin period of time)
+            // For now this could be not a good approch because when
+            // The page is force reloaded the current branch being navigated will be lost, but for now it works.
+            if (isChangedBranch && typeof branch !== 'undefined') {
+                repo.branch = branch;
+            }
             const replacer = ['username', 'reponame', 'branch', 'pullNumber'];
             const repoChanged = JSON.stringify(repo, replacer) !== JSON.stringify(currRepo, replacer);
             if (repoChanged || reload === true) {
               hasError = false;
               $document.trigger(EVENT.REQ_START);
               currRepo = repo;
               treeView.show(repo, token);
+              adapter
+                .getBranches({repo, token})
+                .then((branches) => {
```

Comment:
Look like the rendering tree and branches run parallel because the promise. I think `adapter.getBranches` may not be good approach when the rendering tree work takes 02 job (load tree and load branches). How about changing `adapter.getRepoFromPath` to load branches and append `branches` to returned value

Reply:
`getRepoFromPath` always gets (master), so requesting the tree for master branch while making 2 or more request to get branches (For example in case of too many branches) seems legit or no ?

Reply:
@Fahme Agreed with your concern about using `getRepoFromPath` for loading branch doesn't look right, it's just my random thought to solve the problem about rendering on tree: the tree of file and the branch list display in different timestamp. How about moving `treeView.show(repo, token);` into the `finally` method for promise returned by `getBranches` ?